### PR TITLE
Fix #1817: Remove test:compile dependency on test:fastOptJS.

### LINKED ISSUE #1817: test:compile slowed down by unnecessary fastOptJS
One of the joys of having a rich and expressive type system is that the vast, vast majority of your app can be proven correct just by compilation. This means I call `test:compile` much more than I call `test`. Unfortunately `test:compile` kicks off a `test:fastOptJS` which is very slow in a large project.

Should `fastOptJS` not be deferred until the JS is needed? (i.e. `test` is called)

Some specifics on my project, it has 5 JS modules. A clean `test:compile` takes 171 seconds on an 8-core I7 @ 3.9gz. Running `last test:fastOptJS` on each module shows:

```
[debug] Linker: Read info: 997200 us
[debug] Linker: Compute reachability: 783707 us
[debug] Linker: Assemble LinkedClasses: 1943019 us
[debug] Linker: cache stats: reused: 0 -- invalidated: 8761 -- trees read: 752
[debug] Linker: 3774234 us
[debug] Inc. optimizer: Batch mode: true
[debug] Inc. optimizer: Incremental part: 433060 us
[debug] Inc. optimizer: Optimizing 5266 methods.
[debug] Inc. optimizer: Optimizer part: 2636308 us
[debug] Inc. optimizer: 3123314 us
[debug] Refiner: Compute reachability: 284023 us
[debug] Refiner: Assemble LinkedClasses: 46702 us
[debug] Refiner: 332013 us
[debug] Emitter: Class tree cache stats: reused: 0 -- invalidated: 614
[debug] Emitter: Method tree cache stats: resued: 0 -- invalidated: 3113
[debug] Emitter (write output): 1982288 us

[debug] Linker: Read info: 376171 us
[debug] Linker: Compute reachability: 303091 us
[debug] Linker: Assemble LinkedClasses: 1183697 us
[debug] Linker: cache stats: reused: 0 -- invalidated: 9514 -- trees read: 669
[debug] Linker: 1870131 us
[debug] Inc. optimizer: Batch mode: true
[debug] Inc. optimizer: Incremental part: 109967 us
[debug] Inc. optimizer: Optimizing 4915 methods.
[debug] Inc. optimizer: Optimizer part: 525752 us
[debug] Inc. optimizer: 645502 us
[debug] Refiner: Compute reachability: 275499 us
[debug] Refiner: Assemble LinkedClasses: 20709 us
[debug] Refiner: 296391 us
[debug] Emitter: Class tree cache stats: reused: 0 -- invalidated: 542
[debug] Emitter: Method tree cache stats: resued: 0 -- invalidated: 2842
[debug] Emitter (write output): 767889 us

[debug] Linker: Read info: 437449 us
[debug] Linker: Compute reachability: 374608 us
[debug] Linker: Assemble LinkedClasses: 982423 us
[debug] Linker: cache stats: reused: 0 -- invalidated: 13738 -- trees read: 669
[debug] Linker: 1805915 us
[debug] Inc. optimizer: Batch mode: true
[debug] Inc. optimizer: Incremental part: 1857163 us
[debug] Inc. optimizer: Optimizing 4915 methods.
[debug] Inc. optimizer: Optimizer part: 316309 us
[debug] Inc. optimizer: 2182831 us
[debug] Refiner: Compute reachability: 339724 us
[debug] Refiner: Assemble LinkedClasses: 16334 us
[debug] Refiner: 356251 us
[debug] Emitter: Class tree cache stats: reused: 0 -- invalidated: 542
[debug] Emitter: Method tree cache stats: resued: 0 -- invalidated: 2842
[debug] Emitter (write output): 426643 us

[debug] Linker: Read info: 429108 us
[debug] Linker: Compute reachability: 961366 us
[debug] Linker: Assemble LinkedClasses: 5725820 us
[debug] Linker: cache stats: reused: 0 -- invalidated: 14458 -- trees read: 3939
[debug] Linker: 7126256 us
[debug] Inc. optimizer: Batch mode: true
[debug] Inc. optimizer: Incremental part: 351212 us
[debug] Inc. optimizer: Optimizing 25492 methods.
[debug] Inc. optimizer: Optimizer part: 4708136 us
[debug] Inc. optimizer: 5165959 us
[debug] Refiner: Compute reachability: 650836 us
[debug] Refiner: Assemble LinkedClasses: 49933 us
[debug] Refiner: 700955 us
[debug] Emitter: Class tree cache stats: reused: 0 -- invalidated: 3282
[debug] Emitter: Method tree cache stats: resued: 0 -- invalidated: 17775
[debug] Emitter (write output): 3425674 us

[debug] Linker: Read info: 305000 us
[debug] Linker: Compute reachability: 1184894 us
[debug] Linker: Assemble LinkedClasses: 2090399 us
[debug] Linker: cache stats: reused: 0 -- invalidated: 17519 -- trees read: 5646
[debug] Linker: 3589247 us
[debug] Inc. optimizer: Batch mode: true
[debug] Inc. optimizer: Incremental part: 239366 us
[debug] Inc. optimizer: Optimizing 35478 methods.
[debug] Inc. optimizer: Optimizer part: 7115366 us
[debug] Inc. optimizer: 7381154 us
[debug] Refiner: Compute reachability: 591420 us
[debug] Refiner: Assemble LinkedClasses: 33778 us
[debug] Refiner: 625349 us
[debug] Emitter: Class tree cache stats: reused: 0 -- invalidated: 4755
[debug] Emitter: Method tree cache stats: resued: 0 -- invalidated: 24364
[debug] Emitter (write output): 1695125 us
```

As for the size of my project:

```
     572 text files.
     569 unique files.                                          
       3 files ignored.

http://cloc.sourceforge.net v 1.64  T=0.55 s (1035.6 files/s, 117146.6 lines/s)
-----------------------------------------------------------
                        blank        comment           code
-----------------------------------------------------------
Scala                   11750           4675          47937
-----------------------------------------------------------
```

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Review by @sjrd

File: sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/ScalaJSPluginInternal.scala [hunk 1]
```diff
@@ -575,7 +576,11 @@ object ScalaJSPluginInternal {
           (tf, new ScalaJSFramework(name, env, classpath, logger, console))
         }
       }
-  )
+    ) ++ inConfig(Test)(Seq(
```
Comment (sjrd) [inline sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/ScalaJSPluginInternal.scala]: `inConfig(Test)(Seq(` is redundant. `scalaJSTestFrameworkSettings` itself passes through an `inConfig(Test)` filter down the line.

**Comment by sjrd** [comment]
Maybe it would be worth commenting that incantation to remember what it's supposed to solve when we read it 3 months from now ;-)

That's all.

**Comment by nicolasstucki** [comment]
retest this test

**Comment by nicolasstucki** [comment]
retest this please

File: sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/ScalaJSPluginInternal.scala [hunk 2]
```diff
@@ -574,7 +574,10 @@ object ScalaJSPluginInternal {
         detector.detect(testFrameworks.value) map { case (tf, name) =>
           (tf, new ScalaJSFramework(name, env, classpath, logger, console))
         }
-      }
+      },
+      // Fix #1817
```
Comment (gzm0) [inline sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/ScalaJSPluginInternal.scala]: I think a comment actually explaining what this does would be a good idea, so one doesn't need to have the issue tracker at hand...

Comment (nicolasstucki) [inline sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/ScalaJSPluginInternal.scala]: Agreed

**Comment by nicolasstucki** [comment]
retest this please

**Comment by nicolasstucki** [comment]
retest this please

**Comment by nicolasstucki** [comment]
retest this please

**Comment by sjrd** [comment]
LGTM

**Pull request merged**